### PR TITLE
sleep sort README.md file 

### LINKED ISSUE #2157: Sleep Sort
- [x] C
- [x] C++
- [ ] C#
- [x] CoffeeScript
- [ ] Dart
- [x] Erlang
- [ ] Go - @m-code12
- [x] Java
- [x] Julia
- [ ] JS - @MOHITSAHAL
- [ ] Kotlin
- [ ] Php - @sakshikhachane
- [x] Python
- [x] Readme - @Amitsharma45
- [x] Ruby
- [ ] Rust
- [x] Shell
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Amitsharma45**
Fixes #2157 
#### Checklist:
- [ ] 4 space indentation.
- [ ] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- README.md
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Amitsharma45** [comment]
@jainaman224 @somya-kapoor please review pr

**Comment by somya-kapoor** [comment]
Please add Pseudocode  and Algorithm

**Comment by Amitsharma45** [comment]
@somya-kapoor please check pr .

**Comment by Amitsharma45** [comment]
@somya-kapoor please review pr

File: Sleep_Sort/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            _lk.acquire()
+            print(self.val, end = " ")
+            # Release lock
+            _lk.release()
+
+         def SleepSort(list):
+            ts = []
+            # Intialize a thread corresponding to each element in list
+            for i in list:
+            t = SleepSortThread(i)
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:49]: Please add indentation for statements inside for loop

File: Sleep_Sort/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            ts.append(t)
+            # Start all Threads
+            for i in ts:
+            i.start()
+            # Wait for all threads to terminate
+          for i in ts:
+            i.join()
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:56]: Please follow 4 space indentation here also as we add in code.

File: Sleep_Sort/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+   In general, sleep sort works by starting a separate task for each item to be sorted, where each task sleeps for an interval corresponding to the item's sort key, then emits the item. Items are then collected sequentially in time.
+# Example
+ Let’s assume  we have a laptop that’s  takes 1 seconds to work through each element.
+ 
+ **INPUT: 10 8 6**
+ 
+ - 1s  10
+ - 2s: 8
+ - 3s: 6
+ - 9s:  print(6)  (6  weak up so print it)
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:11]: typo here. Please make it wake up

File: Sleep_Sort/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Sleep Sort
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:1]: You can remove the word In general here. It is the first line of this read me.

File: Sleep_Sort/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ - 10s: print(8)  (8  weak up so print it)
+ - 11s: print(10) (10 weak up so print it)
+ 
+ **OUTPUT : 6 8 10**
+ 
+ # Time complexity
+ - Creating multiple threads is done internally by the OS using a priority queue (used for scheduling purposes). Hence inserting all the  array elements in the priority queue takes O(Nlog N) time.
+- Also the output is obtained only when all the threads are processed, i.e., when all the elements ‘wakes’ up. Since it takes O(arr[i]) time to wake the ith array element’s thread. So it will take a maximum of O(max_element(array)) for the largest element of the array to wake up.
+ - Thus overall time complexitu is ,O(NlogN + max(input)) 
+ ("where, N = number of elements in the input array, and input = input array elements")
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:21]: typo in word complexity

File: Sleep_Sort/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ Let’s assume  we have a laptop that’s  takes 1 seconds to work through each element.
+ 
+ **INPUT: 10 8 6**
+ 
+ - 1s  10
+ - 2s: 8
+ - 3s: 6
+ - 9s:  print(6)  (6  weak up so print it)
+ - 10s: print(8)  (8  weak up so print it)
+ - 11s: print(10) (10 weak up so print it)
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:13]: Please change weak to wake at all places

File: Sleep_Sort/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+         sleep n second
+         print(n)
+      end
+      for arg in args
+         run print(arg) in background 
+      end
+      
+      wait for all processes o finish
+     
+[Courtesy:Quora](https://www.quora.com/What-is-sleep-sort)
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:73]: Please add References section for all the articles you have used while writing this readme. Remove Courtesy.
Also try to add References for genuine and respectable sources and not quora answers.

File: Sleep_Sort/README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+         sleep n second
+         print(n)
+      end
+      for arg in args
+         run print(arg) in background 
+      end
+      
+      wait for all processes o finish
+     
+[Reference:Quora](https://www.geeksforgeeks.org/sleep-sort-king-laziness-sorting-sleeping/)
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:81]: Please give references like this in this Readme: https://github.com/jainaman224/Algo_Ds_Notes/tree/master/Chinese_Remainder_Theorem

Also you have given link for Geeks for geeks and written quora here.
Please provide links to Wikipedia for this.

File: Sleep_Sort/README.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            for i in list:
+               t = SleepSortThread(i)
+               ts.append(t)  
+            # Start all Threads
+            for i in ts:
+               i.start()
+            # Wait for all threads to terminate
+          for i in ts:
+            i.join()
+         x = [2, 4, 3, 1, 6, 8, 4]
```
Comment (singh-shreya6) [inline Sleep_Sort/README.md:63]: Please leave one blank line before input

File: Sleep_Sort/README.md (hunk 8 above)
Comment (Amitsharma45) [reply Sleep_Sort/README.md:81]: ok

**Pull request merged**